Kernel/Process.cpp

```diff
@@ -991,6 +1010,7 @@ int Process::do_exec(NonnullRefPtr<FileDescription> main_program_description, Ve
     m_master_tls_size = master_tls_size;
     m_master_tls_alignment = master_tls_alignment;
 
+    m_pid = new_main_thread->tid();
```

Comment:
How do other systems behave? It seems weird that the result of `getpid()` would change after `exec()`.

Reply:
Linux has the concept of `group_leader`, which in serenity would be the `new_main_thread`. The `group_leader` always has the pid == tid, that's why I did it this way. 

But just noticed that in Linux it's the other way around, it should have been `Thread::current->m_tid = m_pid`, but this isn't possible, since the old `main_thread` may have not exited yet.

Reply:
> Before exec: pid: 8428, tid: 8429
> After  exec: pid: 8428, tid: 8428

This is how Linux behaves with a thread that does exec, it just borrows the group_leader tid.